posix/posix.c

```diff
@@ -2375,7 +2375,7 @@ int posix_waitpid(pid_t child, int *status, int options)
 		}
 		while (!found && (c = c->next) != pinfo->zombies);
 	}
-	while (!found);
+	while (!found && !(options & 1));
```

Comment:
To not use magic numbers in the code, maybe it would be worth to define `WNOHANG` as `1 << 0`

Reply:
@HBuczynski it is defined in libphoenix, I expect this to be solved by @Maxez during posix headers cleanup (it should be defined in kernel's include and used b6 libphoenix)